gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -763,8 +757,12 @@ private void renameDirectoryInternal(
     coopLockOp.ifPresent(
         o -> o.persistAndScheduleRenewal(srcToDstItemNames, srcToDstMarkerItemNames));
     try {
-      // Create the destination directory.
-      mkdir(dst);
+      // Create the destination directory in case Cooperative Locking
+      // is enabled - this is necessary because it uses parent directory
+      // as a marker during recovery of failed rename operations.
```

Comment:
I guess instead of why necessary to create dir for coop lock case, maybe explain why not necessary for non coop lock case.

Reply:
Done.